discretisedfield/field.py

```diff
@@ -2048,21 +2039,68 @@ def pad_fun(vector, pad_width, iaxis, kwargs):
                 index_p1 = dfu.assemble_index(
                     slice(None), 4, {direction_idx: slice(2, None)}
                 )
-                index_0 = dfu.assemble_index(
-                    slice(None), 4, {direction_idx: slice(1, -1)}
-                )
                 index_m1 = dfu.assemble_index(
                     slice(None), 4, {direction_idx: slice(None, -2)}
                 )
                 derivative_array = (
-                    padded_array[index_p1]
-                    - 2 * padded_array[index_0]
-                    + padded_array[index_m1]
-                ) / self.mesh.cell[direction_idx] ** 2
-            else:
-                # The derivative is computed using accuracy of 2 everywhere
-                diff_fd = fd.FinDiff(direction_idx, self.mesh.cell[direction_idx], 2)
-                derivative_array = diff_fd(padded_array)
+                    0.5 * padded_array[index_p1] - 0.5 * padded_array[index_m1]
+                ) / self.mesh.cell[direction_idx]
+
+        elif order == 2:
+            # The derivative is computed using the central difference
+            # with forward/backward difference at the boundaries.
+            if self.mesh.bc == "":
+                if self.mesh.n[direction_idx] < 4:
+                    # Pad with specific values so that the same finite difference
+                    # stencil can be used across the whole array
+                    # central difference = forward difference
+                    # f(1) + f(-1) - 2 f(0) = f(2) + f(0) - 2 f(1)
+                    # f(-1) = f(2) - 3 f(1) + 3f(0)
+
+                    def pad_fun(vector, pad_width, iaxis, kwargs):
+                        if iaxis == direction_idx:
+                            vector[0] = vector[3] - 3 * vector[2] + 3 * vector[1]
+                            vector[-1] = vector[-4] - 3 * vector[-3] + 3 * vector[-2]
```

Comment:
Should have asked this before: When do we need this, i.e. for how many cells? The combination of `n < 4` and index `3` looks "strange".

Reply:
Hopefully I won't explain it here as it will be be good test to see if you can now tell from updated documentation.

Reply:
Makes more sense now. Last question: is this accuracy 1 (instead of the "normal" 2)? Mentioning this would be useful.